packages/aws-cdk-lib/aws-appsync/lib/graphqlapi-base.ts

```diff
@@ -187,7 +187,7 @@ export interface IGraphqlApi extends IResource {
    */
   addRdsDataSource(
     id: string,
-    serverlessCluster: IServerlessCluster,
+    serverlessCluster: IServerlessCluster | IDatabaseCluster,
```

Comment:
This would not be acceptable change. This function is a public API that customers can use. Although it's perfectly fine in typescript, CDK uses JSII to translate typescript to other languages and this specific code change will break JSII compilation unfortunately.

Reply:
made update to have 2 different method to avoid jsii conversion issues.